pkg/expr/mathexp/type_series.go

```diff
@@ -22,70 +23,23 @@ type Series struct {
 	Frame *data.Frame
 }
 
-// SeriesFromFrame validates that the dataframe can be considered a Series type
-// and mutates the frame to be in the format that additional SSE operations expect.
-func SeriesFromFrame(frame *data.Frame) (s Series, err error) {
-	if len(frame.Fields) != 2 {
-		return s, fmt.Errorf("frame must have exactly two fields to be a series, has %v", len(frame.Fields))
+// SeriesFromFrameFields creates a series from the frame fields: one time field and one value field. If value field is not a float64, it tries to convert values to float64.
+// Returns Series created from time and value fields or error if the value field is not a number of indices point to fields of wrong type.
+func SeriesFromFrameFields(frame *data.Frame, timeIdx, valueIdx int) (s Series, e error) {
+	if len(frame.Fields) <= timeIdx {
+		return s, errors.New("cannot find time field")
 	}
-
-	valueIdx := -1
-	timeIdx := -1
-
-	timeNullable := false
-	valueNullable := false
-
-FIELDS:
-	for i, field := range frame.Fields {
-		switch field.Type() {
-		case data.FieldTypeTime:
-			timeIdx = i
-		case data.FieldTypeNullableTime:
-			timeNullable = true
-			timeIdx = i
-		case data.FieldTypeFloat64:
-			valueIdx = i
-		case data.FieldTypeNullableFloat64:
-			valueNullable = true
-			valueIdx = i
-		default:
-			// Handle default case
-			// try to convert to *float64
-			var convertedField *data.Field
-			for j := 0; j < field.Len(); j++ {
-				ff, err := field.NullableFloatAt(j)
-				if err != nil {
-					break
-				}
-				if convertedField == nil { // initialise field
-					convertedField = data.NewFieldFromFieldType(data.FieldTypeNullableFloat64, field.Len())
-					convertedField.Name = field.Name
-					convertedField.Labels = field.Labels
-				}
-				convertedField.Set(j, ff)
-			}
-			if convertedField != nil {
-				frame.Fields[i] = convertedField
-				valueNullable = true
-				valueIdx = i
-			}
-			if valueIdx != -1 && timeIdx != -1 {
-				break FIELDS
-			}
-		}
+	if len(frame.Fields) <= valueIdx {
+		return s, errors.New("cannot find value field")
 	}
 
-	if timeIdx == -1 {
-		return s, fmt.Errorf("no time column found in frame %v", frame.Name)
-	}
-	if valueIdx == -1 {
-		return s, fmt.Errorf("no float64 value column found in frame %v", frame.Name)
-	}
-
-	if timeNullable { // make time not nullable if it is in the input
-		timeSlice := make([]time.Time, 0, frame.Fields[timeIdx].Len())
-		for rowIdx := 0; rowIdx < frame.Fields[timeIdx].Len(); rowIdx++ {
-			val, ok := frame.At(timeIdx, rowIdx).(*time.Time)
+	timeField := frame.Fields[timeIdx]
+	switch timeField.Type() {
+	case data.FieldTypeTime:
```

Comment:
This is a no-op, is that intentional? If it should be both cases, it needs to be comma separated

```suggestion
	case data.FieldTypeTime, data.FieldTypeNullableTime:
```

Reply:
Yes, this is intentional and this should be noop